daemon/daemon.go

```diff
@@ -1142,6 +1142,14 @@ func (daemon *Daemon) GetRemappedUIDGID() (int, int) {
 // created. nil is returned if a child cannot be found. An error is
 // returned if the parent image cannot be found.
 func (daemon *Daemon) ImageGetCached(imgID string, config *runconfig.Config) (*image.Image, error) {
+	// for now just exit if imgID has no children.
+	// maybe parentRefs in graph could be used to store
+	// the Image obj children for faster lookup below but this can
+	// be quite memory hungry.
+	if !daemon.Graph().HasChildren(imgID) {
+		return nil, nil
+	}
+
```

Comment:
Yes, that's what I had in mind. `ByParent` returns a flat list of children instead of a nested map, but it looks like it doesn't matter since `ImageGetCached` just loops over the map keys in the nested map it creates.